### DISCUSSION TITLE
Using librespot in a Swift iOS Application

### DISCUSSION THREAD

**Discussion by AttilaTheFun** [librespot-org/librespot#705] [General]
Has anyone here attempted / been successful with integrating librespot into a Swift iOS application? (Or Objective-C for that matter?)

From this Mozilla article, it sounds possible to use rust libraries on iOS:

https://mozilla.github.io/firefox-browser-architecture/experiments/2017-09-06-rust-on-ios.html

It sounds like I would need to link the library after compiling it for arm-64 and add a bridging header.

**Comment by Johannesd3** [comment]
Looks as if you need C bindings. If you want to implement them yourself, please take a look at #689.
